lifetimes/utils.py

```diff
@@ -110,31 +110,25 @@ def find_first_transactions(transactions, customer_id_col, datetime_col, monetar
     if monetary_value_col:
         select_columns.append(monetary_value_col)
 
-    transactions = transactions[select_columns].sort(select_columns).copy()
+    transactions = transactions[select_columns].sort_values(select_columns).copy()
 
     # make sure the date column uses datetime objects, and use Pandas' DateTimeIndex.to_period()
     # to convert the column to a PeriodIndex which is useful for time-wise grouping and truncating
     transactions[datetime_col] = pd.to_datetime(transactions[datetime_col], format=datetime_format)
     transactions = transactions.set_index(datetime_col).to_period(freq)
 
-    transactions = transactions.ix[(transactions.index <= observation_period_end)].reset_index()
-
-    period_groupby = transactions.groupby([datetime_col, customer_id_col], sort=False, as_index=False)
+    transactions = transactions.loc[(transactions.index <= observation_period_end)].reset_index()
 
     if monetary_value_col:
         # when we have a monetary column, make sure to sum together any values in the same period
+        period_groupby = transactions.groupby([datetime_col, customer_id_col], sort=False, as_index=False)
         period_transactions = period_groupby.sum()
     else:
-        # by calling head() on the groupby object, the datetime_col and customer_id_col columns
-        # will be reduced
-        period_transactions = period_groupby.head(1)
+        # drop duplicated rows
+        period_transactions = transactions.drop_duplicates()
 
     # initialize a new column where we will indicate which are the first transactions
-    period_transactions['first'] = False
-    # find all of the initial transactions and store as an index
-    first_transactions = period_transactions.groupby(customer_id_col, sort=True, as_index=False).head(1).index
-    # mark the initial transactions as True
-    period_transactions.loc[first_transactions, 'first'] = True
+    period_transactions['first'] = ~(period_transactions.duplicated(customer_id_col))
```

Comment:
Because when `duplicated` saw row and didn't see it yet it marks as False. Then we doing `~` as opposite. I did that because I thought that it'll be faster... But benchmarks show no speed improvement. May be it's better to revert to previous with so many aspects of `duplicated` and `drop_duplicates`.